### ISSUE TITLE
.NET 7 Native AOT error: Generic virtual method pointer lookup failure

### ISSUE THREAD

**Issue by littlehoneybunnytuttifruttipumpkinpie** [Cysharp/MemoryPack#75]
First, thanks for updating MemoryPack to work with .NET 7 and AOT.
Unfortunately my native AOT test program (latest .NET 7) is crashing with the error below:
![image](https://user-images.githubusercontent.com/20084989/204111373-34cbba8d-a870-4047-a862-a024dc088942.png)

(it works OK in debug and release mode in Visual Studio, but crashes only after publishing and running the AOT executable)

Here is the full test code:
```
using MemoryPack;

namespace Tests
{
    [MemoryPackable]
    public partial class MemPackObject
    {
        public string TestString { get; set; }
    }

    class MemPackTest
    {
        static void Main(string[] args)
        {
            MemPackObject test_object = new MemPackObject();
            test_object.TestString = "ABC";
            Console.WriteLine("Initial value before serializing: " + test_object.TestString);

            byte[] serialized_object = MemoryPackSerializer.Serialize(test_object);
            MemPackObject? deserialized_object = MemoryPackSerializer.Deserialize<MemPackObject>(serialized_object);

            if (deserialized_object != null)
            {
                Console.WriteLine("Final value after deserializing: " + deserialized_object.TestString);
            }

            Console.WriteLine("Test completed. Press any key to exit");
            Console.ReadKey();
        }

    }
}
```

**Comment by neuecc** [comment]
Thanks.
I suspect it is a bug in the Native AOT compiler, as I reproduced it even with minimized code.
Reported at https://github.com/dotnet/runtime/issues/78882

**Comment by neuecc** [comment]
I received a response.
It seems to be a bug and will not be resolved until .NET 8.

> The workaround is to tell the compiler manually that the target needs to be generated. It is not pretty. Pull request https://github.com/dotnet/runtime/pull/78904 adds support for at least telling what needs to be pregenerated and I'm hoping we can service that part in January for 7.0.

> It's likely the root cause of the problem will only be fixed in 8.0. Sorry, it's a very risky fix.

I have noted this in the ReadMe.

**Comment by GerardSmit** [comment]
This has been fixed in https://github.com/dotnet/runtime/pull/80601

It's possible to use the daily build of `Microsoft.DotNet.ILCompiler` in .NET 7.0.  
**Note that this is a daily build so it's possible there are other bugs.**

1. Create `nuget.config` in your solution (where the .sln-file is located) with the following content:
    ```xml
    <?xml version="1.0" encoding="utf-8"?>
    <configuration>
      <packageSources>
        <add key="dotnet8" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet8/nuget/v3/index.json" />
      </packageSources>
    </configuration>
    ```
2. Add the following to your `.csproj` (change win-x64 if you're not using Windows):
    ```xml
    <PackageReference Include="Microsoft.DotNet.ILCompiler; runtime.win-x64.Microsoft.DotNet.ILCompiler" Version="8.0.0-alpha.1.23067.11" />
    ```

After that you can build the project with AOT (e.g. `dotnet publish -r win-x64 -c Release -p PublishAot=True`). If you run the application it should give the expected result:

```
Initial value before serializing: ABC
Final value after deserializing: ABC
Test completed. Press any key to exit
```
